server/play-server/src/test/scala/sttp/tapir/server/play/PlayServerTest.scala

```diff
@@ -98,7 +109,61 @@ class PlayServerTest extends TestSuite {
                 }
             }
             .unsafeToFuture()
-        }
+        },
+        Test("chunked transmission lasts longer than given timeout") {
+          val chunkSize = 100
+          val beforeSendingSecondChunk: FiniteDuration = 2.second
+          val requestTimeout: FiniteDuration = 1.second
+
+          val e =
+            endpoint.post
+              .in(header[Long](HeaderNames.ContentLength))
+              .in(streamTextBody(PekkoStreams)(CodecFormat.TextPlain()))
+              .out(header[Long](HeaderNames.ContentLength))
+              .out(streamTextBody(PekkoStreams)(CodecFormat.TextPlain()))
+              .serverLogicSuccess[Future] { case (length, stream) =>
+                Future.successful(length, stream)
+              }
+
+          val components: DefaultPekkoHttpServerComponents = new DefaultPekkoHttpServerComponents {
+            val initialServerConfig: ServerConfig = ServerConfig(port = Some(0), address = "127.0.0.1", mode = Mode.Test)
+
+            val customConf: Configuration =
+              Configuration(
+                ConfigFactory.parseString(s"play { server.pekko.requestTimeout=${requestTimeout.toString} }")
+              )
+            override lazy val serverConfig: ServerConfig =
+              initialServerConfig.copy(configuration = customConf.withFallback(initialServerConfig.configuration))
+            override lazy val actorSystem: ActorSystem = ActorSystem("tapir", defaultExecutionContext = Some(_actorSystem.dispatcher))
+            override lazy val router: Router = Router.from(PlayServerInterpreter().toRoutes(e)).withPrefix("/chunks")
+          }
+
+          def createStream(chunkSize: Int, beforeSendingSecondChunk: FiniteDuration): Stream[IO, Byte] = {
+            val chunk = Chunk.array(Array.fill(chunkSize)('A'.toByte))
+            val initialChunks = Stream.chunk(chunk)
+            val delayedChunk = Stream.sleep[IO](beforeSendingSecondChunk) >> Stream.chunk(chunk)
+            initialChunks ++ delayedChunk
+          }
+
+          val inputStream = createStream(chunkSize, beforeSendingSecondChunk)
+
+          val bind = IO.blocking(components.server)
+          Resource.make(bind)(s => IO.blocking(s.stop()))
+            .map(_.mainAddress.getPort)
+            .use { port =>
+              basicRequest
+                .post(uri"http://localhost:$port/chunks")
+                .contentLength(2 * chunkSize)
+                .streamBody(Fs2Streams[IO])(inputStream)
+                .send(backend)
+                .map{ response =>
+                  response.code shouldBe StatusCode.Ok
```

Comment:
hm well if this test passes, something is wrong - we set the timeout to 1s, so we should never receive a response if it takes 2s to send it? unless the request timeout is for something else? 

anyway, this doesn't test the scenario from the test case - where the transmission is interrupted half-way because of connection problems; I don't know if we can simulate this in a test case, but using a timeout is a good approximation. But probably a good way to check if we can at all reproduce the bug is to run: a long-running client sender process; a server process; then `kill -9` the client process when it's half-way sending the data, and seeing on the server if received the incomplete data in the server logic

Reply:
I have given it another try as you suggested. There are [playServer](https://github.com/softwaremill/tapir/pull/4214/commits/9402d02a2f735e0c553132f6c6d62687dd0dd183#diff-680e4d576aafc5630a9032b9619d0f1b5d93d8d6f55871da044e8eca33d52897) and [longLastingClient](https://github.com/softwaremill/tapir/pull/4214/commits/9402d02a2f735e0c553132f6c6d62687dd0dd183#diff-91749d496b59d0d54eacb34e5cecb106ecb4495584408b87ce58f4baad8340c4) but I don't know what's wrong with that approach. Suggestions are welcome 💡

Reply:
When you write "I don't know what's wrong with this approach ", do you mean that it works as expected (that is: you run both, interrupt the client brutally after some time, and the server properly closes the connection), or is there something else that's wrong?

Reply:
I'm trying with the following steps:

1. `$ scala-cli run playServer.scala`

```
[playServer-pekko.actor.default-dispatcher-4] INFO org.apache.pekko.event.slf4j.Slf4jLogger - Slf4jLogger started
[application-pekko.actor.default-dispatcher-6] INFO org.apache.pekko.event.slf4j.Slf4jLogger - Slf4jLogger started
[main] INFO play.api.Play - Application started (Dev) (no global state)
[main] INFO play.core.server.NettyServer - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
```

2. `$ scala-cli run longLastingClient.scala`

```
ByteString(65, 65, 65, 65, 65, 65, 65, 65, 65, 65) 17:46:14.514330
ByteString(65, 65, 65, 65, 65, 65, 65, 65, 65, 65) 17:46:15.534314
ByteString(65, 65, 65, 65, 65, 65, 65, 65, 65, 65) 17:46:16.553033
ByteString(65, 65, 65, 65, 65, 65, 65, 65, 65, 65) 17:46:17.573135
....
```

3. server side

```
Received 10000 bytes, Source(SourceShape(Map.out(1467551936))) bytes in total
```

4.  `$ ps aux | grep longLastingClient | awk '{print $2}' | head -n 1 | xargs kill -9`

5.  and nothing new (error/exception) on server side 🤔